alibi/explainers/partial_dependence.py

```diff
@@ -0,0 +1,1294 @@
+            pd_values=pd_values,
+            feature_deciles=feature_deciles,
+        )
+        data['meta'].update(
+            feature_names=self.feature_names,
+            categorical_names=self.categorical_names,
+            target_names=self.target_names,
+            response_method=response_method,
+            method=method,
+            kind=kind
+        )
+        return Explanation(meta=copy.deepcopy(self.meta), data=data)
+
+    def reset_predictor(self, predictor: Any) -> None:
+        """
+        Resets the predictor function
+
+        Parameters
+        ----------
+        predictor
+            New `sklearn` estimator.
+        """
+        self.predictor = predictor
+
+
+# No type check due to the generic explanation object
+@no_type_check
+def plot_pd(exp: Explanation,
+            features_list: Union[List[int], Literal['all']] = 'all',
+            target_idx: int = 0,
+            n_cols: int = 3,
+            n_ice: Union[str, int, List[int]] = 'all',
+            centered: bool = True,
+            levels: int = 8,
+            ax: Optional[Union['plt.Axes', np.ndarray]] = None,
+            sharey: str = 'all',
+            pd_num_kw: Optional[dict] = None,
+            ice_num_kw: Optional[dict] = None,
+            pd_cat_kw: Optional[dict] = None,
+            ice_cat_kw: Optional[dict] = None,
+            pd_num_num_kw: Optional[dict] = None,
+            pd_num_cat_kw: Optional[dict] = None,
+            pd_cat_cat_kw: Optional[dict] = None,
+            fig_kw: Optional[dict] = None,
+            seed: Optional[int] = None) -> 'np.ndarray':
+    """
+    Plot partial dependence curves on matplotlib axes.
+
+    Parameters
+    ----------
+    exp
+        An `Explanation` object produced by a call to the
+        :py:meth:`alibi.explainers.partial_dependence.PartialDependence.explain` method.
+    features_list
+        A list of features for which to plot the partial dependence curves or ``'all'`` for all features.
+        Can be an integers denoting feature index denoting entries in `exp.feature_names`. Defaults to ``'all'``.
+    target_idx
+        Target index for which to plot the partial dependence (PD) curves. Can be a mix of integers denoting target
+        index or strings denoting entries in `exp.meta['target_names']`.
+    n_cols
+        Number of columns to organize the resulting plot into.
+    n_ice
+        Number of ICE plots to be displayed. Can be
+
+         - a string taking the value ``'all'`` to display the ICE curves for every instance in the reference dataset.
+
+         - an integer for which `n_ice` instances from the reference dataset will be sampled uniformly at random to
+           display their ICE curves.
+
+         - a list of integers, where each integer represents an index of an instance in the reference dataset to
+           display their ICE curves.
+
+    centered
+        Boolean flag to center the individual conditional expectation (ICE) curves.
```

Comment:
Would be useful to add another sentence why this is useful to do.

Reply:
I've included an explanation.